crates/tx5-connection/src/webrtc/go_pion.rs

```diff
@@ -269,3 +316,94 @@ fn spawn_data_chan(
         }),
     )))
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    async fn conn_dropped_on_peer_connection_state_change(
+        state: tx5_go_pion::PeerConnectionState,
+    ) {
+        let (cmd_send, cmd_recv) = CloseSend::sized_channel(1024);
+        let (evt_send, _evt_recv) = CloseSend::sized_channel(1024);
+        let (peer_evt_send, peer_evt_recv) =
+            tokio::sync::mpsc::unbounded_channel();
+
+        // Create PeerConnection
+        let config = tx5_go_pion::PeerConnectionConfig::default();
+        let (peer, _real_peer_events) =
+            tx5_go_pion::PeerConnection::new(config)
+                .await
+                .unwrap();
+
+        // Spawn the task_inner
+        let task_handle = tokio::spawn({
+            let cmd_send_clone = cmd_send.clone();
+            async move {
+                task_inner(
+                    true, // is_polite
+                    1024, // send_buffer
+                    evt_send,
+                    cmd_send_clone,
+                    cmd_recv,
+                    peer,
+                    peer_evt_recv,
+                )
+                .await
+            }
+        });
+
+        // Send connected state
+        peer_evt_send
+            .send(tx5_go_pion::PeerConnectionEvent::State(
+                tx5_go_pion::PeerConnectionState::Connected,
+            ))
+            .unwrap();
+
+        assert!(!task_handle.is_finished());
```

Comment:
Shouldn't we in principle wait for a while here before checking, since sending over the channel may take a while?

Reply:
I'm don't think either approach (including mine) really asserts anything reliably -- it takes some unknown time for the peer connection state change to be handled, so with either approach we're still just asserting "the connection is still open at this time" -- but it could still close after which would not cause the assertion to fail. I don't know a correct way to test this.

Reply:
Deleted the assertion